:construction: Delete table metadata versioning

mitchdawson1982 opened the pull request:
This PR moves much of the existing delete table code out into the base package.

Commit:
:construction: Delete table metadata versioning

Commit:
:truck: Move delete glue table funcion to its own base module

Commit:
:white_check_mark: Refactor delete table tests

Commit:
:fire: Refactor and remove code to VersionCreator class

Commit:
:sparkles: Now handles removing tables, data files, versioning and copying

Commit:
:sparkles: Added new and refactored utility functions

Commit:
:fire: Remove unused fixture

Commit:
:recycle: Simplify update_metadata_remove_schemas

Commit:
:necktie: Validate on schema list rather than presence of schema key

Commit:
:white_check_mark: Add test to validate deletion of schema.json file

Commit:
:loud_sound: Update success message

Commit:
:fire: Remove create table function as not needed at this time

Commit:
:white_check_mark: Updated spec folder version

Commit:
:memo: Update python base change log and config with new version

Commit:
:memo: Update python base change log and config with new version

Commit:
:test_tube: Fix python base tests to use s3_client

Commit:
:white_check_mark: Add tests for VersionCreator.update_metadata_remove_schema method

Commit:
:memo: Update change logs and bump delete table version

Commit:
:memo: Update change logs and bump delete table version

Commit:
:memo: Update change logs and bump delete table version

Commit:
:memo: Update change logs and bump delete table version

Commit:
:heavy_plus_sign: Add pyparsing===3.1.1

mitchdawson1982 committed:
Update delete_table.py update success response with format_response_json

Commit:
:white_check_mark: Update test to reflect success response object change

dependabot[bot] committed:
github-actions(deps): bump bridgecrewio/checkov-action

Bumps [bridgecrewio/checkov-action](https://github.com/bridgecrewio/checkov-action) from 12.2559.0 to 12.2562.0.
- [Release notes](https://github.com/bridgecrewio/checkov-action/releases)
- [Commits](https://github.com/bridgecrewio/checkov-action/compare/7930f52ee6bd57403a7918f765291c72c81a7ba8...21878105b7c7863d7ec460bfc96e026150b2b633)

---
updated-dependencies:
- dependency-name: bridgecrewio/checkov-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
github-actions(deps): bump actions/dependency-review-action

Bumps [actions/dependency-review-action](https://github.com/actions/dependency-review-action) from 3.1.0 to 3.1.1.
- [Release notes](https://github.com/actions/dependency-review-action/releases)
- [Commits](https://github.com/actions/dependency-review-action/compare/6c5ccdad469c9f8a2996bfecaec55a631a347034...9f45b2463b475767b61721ccfef113fef513e6aa)

---
updated-dependencies:
- dependency-name: actions/dependency-review-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
terraform(deps): bump terraform-aws-modules/eks/aws

Bumps [terraform-aws-modules/eks/aws](https://github.com/terraform-aws-modules/terraform-aws-eks) from 19.18.0 to 19.19.0.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-eks/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-eks/compare/v19.18.0...v19.19.0)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/eks/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
terraform(deps): bump terraform-aws-modules/iam/aws

Bumps [terraform-aws-modules/iam/aws](https://github.com/terraform-aws-modules/terraform-aws-iam) from 5.30.0 to 5.30.1.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-iam/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-iam/compare/v5.30.0...v5.30.1)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/iam/aws
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

LavMatt commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -198,25 +199,68 @@ def extract_timestamp_from_curated_path(string: str):
+
+
+def delete_all_element_version_data_files(data_product_name: str, table_name: str):
+    """Deletes raw and curated data for all element versions"""
```
To me this doesn't so much feel like a pathing related function but more a versioning related function

mitchdawson1982 replied:
Function now moved to the versioning module.

LavMatt commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -198,25 +199,68 @@ def extract_timestamp_from_curated_path(string: str):
+    element = DataProductElement.load(
+        element_name=table_name, data_product_name=data_product_name
+    )
+    raw_prefixes = generate_all_element_version_prefixes(
```
this returns all versions as per the metadata, and that won;t match to the data, e.g. you won;t have 1.1, 1.2, but only major versions for data

mitchdawson1982 replied:
Agreed that this will be revisited once data versioning has been implemented.

LavMatt commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -198,25 +199,68 @@ def extract_timestamp_from_curated_path(string: str):
+    raw_prefixes = generate_all_element_version_prefixes(
+        "raw", data_product_name, table_name
+    )
+    curated_prefixes = generate_all_element_version_prefixes(
```
same for this. Maybe we could just infer all major versions from what's returned

mitchdawson1982 replied:
Agreed that this will be revisited once data versioning has been implemented.

LavMatt commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -198,25 +199,68 @@ def extract_timestamp_from_curated_path(string: str):
+        "curated", data_product_name, table_name
+    )
+
+    s3_recursive_delete(element.data_product.raw_data_bucket, raw_prefixes)
```
Seems to me slightly conflicting behaviour deleting from an existing version without any upversioning of database but still upversioning metadata. This is where a user no longer wants any of the data available in any versions? I'm just wondering how if we have this and we have a `remove a schema` endpoint how a user would have to careful in choosing which one to use. Whereas `remove a schema` would remove it from say a v2 onwards.

Later we're creating a new major version metadata but not a new version of the data. Having reread #1662 it says both to delete raw and curated which is happening but also to not delete raw.

I'm wondering if it was expected to have a separate delete schema/table with versioning endpoint or if this is to be the only delete

mitchdawson1982 replied:
Agreed that this will be revisited once data and database versioning has been implemented.

LavMatt reviewed (commented):
This mostly looks good Mitch.

It will delete data and major upversion metadata but (and it may be me not following the intention of the ticket) i'm not 100% convinced we should be deleting data from within versions, e.g. data_product_v1 has tbl1 and tbl2 and then after calling this endpoint to delete tbl2, the same data_product_v1 only has tbl1.

And not sure having data product data version and metadata versions misalign is correct.

Another observation i made about getting all versions to delete data using versions from metadata, which will return all minor versions when data are versioned at major level only.

Commit:
:ambulance: Update repository name

mitchdawson1982 committed:
Merge branch 'main' into dpl-1662-metadata-versioning

Commit:
:bookmark: Update changelog and increase version

dependabot[bot] committed:
terraform(deps): bump terraform-aws-modules/iam/aws from 5.30.0 to 5.30.1 in /terraform/aws/analytical-platform-data-production/airflow (#2222)

terraform(deps): bump terraform-aws-modules/iam/aws

Bumps [terraform-aws-modules/iam/aws](https://github.com/terraform-aws-modules/terraform-aws-iam) from 5.30.0 to 5.30.1.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-iam/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-iam/compare/v5.30.0...v5.30.1)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/iam/aws
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

PriyaBasker23 committed:
Updated docs (#2241)

* Updated docs

* Added 400 response

dependabot[bot] committed:
github-actions(deps): bump bridgecrewio/checkov-action

Bumps [bridgecrewio/checkov-action](https://github.com/bridgecrewio/checkov-action) from 12.2562.0 to 12.2564.0.
- [Release notes](https://github.com/bridgecrewio/checkov-action/releases)
- [Commits](https://github.com/bridgecrewio/checkov-action/compare/21878105b7c7863d7ec460bfc96e026150b2b633...b74b4731be4b3ac56f6a419aa8345b4884a5b27b)

---
updated-dependencies:
- dependency-name: bridgecrewio/checkov-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
♻️ Refactor GitHub Terraform component (#1985)

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>

dependabot[bot] committed:
terraform(deps): bump hashicorp/aws

Bumps [hashicorp/aws](https://github.com/hashicorp/terraform-provider-aws) from 5.23.0 to 5.24.0.
- [Release notes](https://github.com/hashicorp/terraform-provider-aws/releases)
- [Changelog](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-provider-aws/compare/v5.23.0...v5.24.0)

---
updated-dependencies:
- dependency-name: hashicorp/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
github-actions(deps): bump aquasecurity/trivy-action

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.13.1 to 0.14.0.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/f78e9ecf42a1271402d4f484518b9313235990e1...2b6a709cf9c4025c5438138008beaddbb02086f0)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
terraform(deps): bump hashicorp/aws

Bumps [hashicorp/aws](https://github.com/hashicorp/terraform-provider-aws) from 5.23.1 to 5.24.0.
- [Release notes](https://github.com/hashicorp/terraform-provider-aws/releases)
- [Changelog](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-provider-aws/compare/v5.23.1...v5.24.0)

---
updated-dependencies:
- dependency-name: hashicorp/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
terraform(deps): bump integrations/github in /terraform/github

Bumps [integrations/github](https://github.com/integrations/terraform-provider-github) from 5.41.0 to 5.42.0.
- [Release notes](https://github.com/integrations/terraform-provider-github/releases)
- [Changelog](https://github.com/integrations/terraform-provider-github/blob/main/CHANGELOG.md)
- [Commits](https://github.com/integrations/terraform-provider-github/compare/v5.41.0...v5.42.0)

---
updated-dependencies:
- dependency-name: integrations/github
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
terraform(deps): bump hashicorp/aws

Bumps [hashicorp/aws](https://github.com/hashicorp/terraform-provider-aws) from 5.21.0 to 5.24.0.
- [Release notes](https://github.com/hashicorp/terraform-provider-aws/releases)
- [Changelog](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-provider-aws/compare/v5.21.0...v5.24.0)

---
updated-dependencies:
- dependency-name: hashicorp/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
✨ Ways of Working 2.0 (#2153)

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>
Co-authored-by: Jake H-P <72385707+jhpyke@users.noreply.github.com>
Co-authored-by: Jacob Hamblin-Pyke <jacob.hamblin-pyke@digital.justice.gov.uk>
Co-authored-by: Michael Collins <15347726+michaeljcollinsuk@users.noreply.github.com>

dependabot[bot] committed:
terraform(deps): bump terraform-aws-modules/iam/aws

Bumps [terraform-aws-modules/iam/aws](https://github.com/terraform-aws-modules/terraform-aws-iam) from 5.30.0 to 5.30.1.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-iam/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-iam/compare/v5.30.0...v5.30.1)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/iam/aws
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

tom-webber committed:
Update Data Product tech docs (#2177)

* add versioning documentation
* update spec and governance docs with optional implemented fields
* add erb headers for previously added doc pages
* Update docs/source/documentation/data-products/components/product-specification.html.md.erb
* clarify `email` contact info definition

mitchdawson1982 committed:
Swagger docs update (#2250)

* :memo: Update delete table swagger documentation

* :memo: Update docs for presigned url and delete table

* terraform(deps): bump terraform-aws-modules/iam/aws from 5.30.0 to 5.30.1 in /terraform/aws/analytical-platform-data-production/airflow (#2222)

terraform(deps): bump terraform-aws-modules/iam/aws

Bumps [terraform-aws-modules/iam/aws](https://github.com/terraform-aws-modules/terraform-aws-iam) from 5.30.0 to 5.30.1.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-iam/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-iam/compare/v5.30.0...v5.30.1)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/iam/aws
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* terraform(deps): bump terraform-aws-modules/iam/aws

Bumps [terraform-aws-modules/iam/aws](https://github.com/terraform-aws-modules/terraform-aws-iam) from 5.30.0 to 5.30.1.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-iam/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-iam/compare/v5.30.0...v5.30.1)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/iam/aws
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* terraform(deps): bump terraform-aws-modules/eks/aws

Bumps [terraform-aws-modules/eks/aws](https://github.com/terraform-aws-modules/terraform-aws-eks) from 19.18.0 to 19.19.0.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-eks/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-eks/compare/v19.18.0...v19.19.0)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/eks/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* github-actions(deps): bump actions/dependency-review-action

Bumps [actions/dependency-review-action](https://github.com/actions/dependency-review-action) from 3.1.0 to 3.1.1.
- [Release notes](https://github.com/actions/dependency-review-action/releases)
- [Commits](https://github.com/actions/dependency-review-action/compare/6c5ccdad469c9f8a2996bfecaec55a631a347034...9f45b2463b475767b61721ccfef113fef513e6aa)

---
updated-dependencies:
- dependency-name: actions/dependency-review-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* github-actions(deps): bump bridgecrewio/checkov-action

Bumps [bridgecrewio/checkov-action](https://github.com/bridgecrewio/checkov-action) from 12.2559.0 to 12.2562.0.
- [Release notes](https://github.com/bridgecrewio/checkov-action/releases)
- [Commits](https://github.com/bridgecrewio/checkov-action/compare/7930f52ee6bd57403a7918f765291c72c81a7ba8...21878105b7c7863d7ec460bfc96e026150b2b633)

---
updated-dependencies:
- dependency-name: bridgecrewio/checkov-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Updated docs (#2241)

* Updated docs

* Added 400 response

* ♻️ Refactor GitHub Terraform component (#1985)

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>

* terraform(deps): bump hashicorp/aws

Bumps [hashicorp/aws](https://github.com/hashicorp/terraform-provider-aws) from 5.23.1 to 5.24.0.
- [Release notes](https://github.com/hashicorp/terraform-provider-aws/releases)
- [Changelog](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-provider-aws/compare/v5.23.1...v5.24.0)

---
updated-dependencies:
- dependency-name: hashicorp/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* terraform(deps): bump hashicorp/aws

Bumps [hashicorp/aws](https://github.com/hashicorp/terraform-provider-aws) from 5.23.0 to 5.24.0.
- [Release notes](https://github.com/hashicorp/terraform-provider-aws/releases)
- [Changelog](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-provider-aws/compare/v5.23.0...v5.24.0)

---
updated-dependencies:
- dependency-name: hashicorp/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* terraform(deps): bump hashicorp/aws

Bumps [hashicorp/aws](https://github.com/hashicorp/terraform-provider-aws) from 5.21.0 to 5.24.0.
- [Release notes](https://github.com/hashicorp/terraform-provider-aws/releases)
- [Changelog](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-provider-aws/compare/v5.21.0...v5.24.0)

---
updated-dependencies:
- dependency-name: hashicorp/aws
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* ✨ Ways of Working 2.0 (#2153)

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>
Co-authored-by: Jake H-P <72385707+jhpyke@users.noreply.github.com>
Co-authored-by: Jacob Hamblin-Pyke <jacob.hamblin-pyke@digital.justice.gov.uk>
Co-authored-by: Michael Collins <15347726+michaeljcollinsuk@users.noreply.github.com>

* github-actions(deps): bump aquasecurity/trivy-action

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.13.1 to 0.14.0.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/f78e9ecf42a1271402d4f484518b9313235990e1...2b6a709cf9c4025c5438138008beaddbb02086f0)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* github-actions(deps): bump bridgecrewio/checkov-action

Bumps [bridgecrewio/checkov-action](https://github.com/bridgecrewio/checkov-action) from 12.2562.0 to 12.2564.0.
- [Release notes](https://github.com/bridgecrewio/checkov-action/releases)
- [Commits](https://github.com/bridgecrewio/checkov-action/compare/21878105b7c7863d7ec460bfc96e026150b2b633...b74b4731be4b3ac56f6a419aa8345b4884a5b27b)

---
updated-dependencies:
- dependency-name: bridgecrewio/checkov-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* terraform(deps): bump integrations/github in /terraform/github

Bumps [integrations/github](https://github.com/integrations/terraform-provider-github) from 5.41.0 to 5.42.0.
- [Release notes](https://github.com/integrations/terraform-provider-github/releases)
- [Changelog](https://github.com/integrations/terraform-provider-github/blob/main/CHANGELOG.md)
- [Commits](https://github.com/integrations/terraform-provider-github/compare/v5.41.0...v5.42.0)

---
updated-dependencies:
- dependency-name: integrations/github
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* terraform(deps): bump terraform-aws-modules/iam/aws

Bumps [terraform-aws-modules/iam/aws](https://github.com/terraform-aws-modules/terraform-aws-iam) from 5.30.0 to 5.30.1.
- [Release notes](https://github.com/terraform-aws-modules/terraform-aws-iam/releases)
- [Changelog](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terraform-aws-modules/terraform-aws-iam/compare/v5.30.0...v5.30.1)

---
updated-dependencies:
- dependency-name: terraform-aws-modules/iam/aws
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update Data Product tech docs (#2177)

* add versioning documentation
* update spec and governance docs with optional implemented fields
* add erb headers for previously added doc pages
* Update docs/source/documentation/data-products/components/product-specification.html.md.erb
* clarify `email` contact info definition

* :memo: Update changelog & version

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: PriyaBasker23 <90777804+PriyaBasker23@users.noreply.github.com>
Co-authored-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>
Co-authored-by: Jake H-P <72385707+jhpyke@users.noreply.github.com>
Co-authored-by: Jacob Hamblin-Pyke <jacob.hamblin-pyke@digital.justice.gov.uk>
Co-authored-by: Michael Collins <15347726+michaeljcollinsuk@users.noreply.github.com>
Co-authored-by: Tom Webber <80110358+tom-webber@users.noreply.github.com>

tom-webber committed:
Dp 1797 update athena_load to use existing schema for curated table (#2156)

* Use existing schema for curated table if exists
* test load handler with and without existing schema
* refactor athena_load_handler tests
* database name for DataProductElement.curated_data_table includes latest major version
* trim schema table desc and col comment to glue limits
* clear defaults from glue schema template
* add `database_name` to `BaseJsonSchema` - used in glue schema
* add `database_name` to `DataProductElement` - used in `curated_data_table`

MatMoore committed:
Check csv for issues even if schema is not available (#2205)

* Check csv for issues even if schema is not available

If the schema is available, we can validate column types.
However, even if we don't have a schema, we need to check that
the csv looks parsable and does not contain embedded newlines.

* Changelog and versioning

* Use the correct name for logging to s3

MatMoore committed:
Add data product name to the logger and update to latest base image in reload/resync lambdas (#2211)

* Add data product name to the logger where missing

* Upgrade to latest base image

This picks up the new path format changes.

Commit:
🚑 Add `limit-severities-for-sarif` option to `aquasecurity/trivy-action` (#2254)

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>

mitchdawson1982 committed:
Remove delete table schema references (#2257)

* :fire: Remove delete table schema references

* :memo: Update changelog and config.json for v1.0.11

dependabot[bot] committed:
github-actions(deps): bump actions/dependency-review-action

Bumps [actions/dependency-review-action](https://github.com/actions/dependency-review-action) from 3.1.1 to 3.1.2.
- [Release notes](https://github.com/actions/dependency-review-action/releases)
- [Commits](https://github.com/actions/dependency-review-action/compare/9f45b2463b475767b61721ccfef113fef513e6aa...fde92acd0840415674c16b39c7d703fc28bc511e)

---
updated-dependencies:
- dependency-name: actions/dependency-review-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
github-actions(deps): bump bridgecrewio/checkov-action

Bumps [bridgecrewio/checkov-action](https://github.com/bridgecrewio/checkov-action) from 12.2564.0 to 12.2566.0.
- [Release notes](https://github.com/bridgecrewio/checkov-action/releases)
- [Commits](https://github.com/bridgecrewio/checkov-action/compare/b74b4731be4b3ac56f6a419aa8345b4884a5b27b...61c04ede9927c5a4daf5dfaa4a98ae5fd026aa96)

---
updated-dependencies:
- dependency-name: bridgecrewio/checkov-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

tom-webber committed:
fix container dockerfiles: `LAMDA_TASK_ROOT` -> `LAMBDA_TASK_ROOT` (#2253)

* fix container dockerfiles: `LAMDA_TASK_ROOT` -> `LAMBDA_TASK_ROOT`
* update `evaluate_checks` workflow error reporting

MatMoore committed:
Return text/plain preview instead of application/json (#2260)

Currently, API gateway is returning the response as a JSON encoded
string, e.g.

`"| col_0              | col_1               | col_2               | col_3               | extraction_timestamp |\n| 0.1915194503788923 | 0.3648859839013723  | 0.0598092227798519  | 0.2852509600245098  | 20231023T144052Z     |\n| 0.7853585837137692 | 0.3688240060019745  | 0.6748809435823302  | 0.19567517866589823 | 20231023T144052Z     |\n| 0.7799758081188035 | 0.9331401019825216  | 0.5946247799344488  | 0.38231745203150647 | 20231023T144052Z     |\n| 0.2725926052826416 | 0.6513781432265774  | 0.5333101629987506  | 0.05387368514623658 | 20231023T144052Z     |\n| 0.2764642551430967 | 0.3972025777261542  | 0.04332406269480349 | 0.45164840826085906 | 20231023T144052Z     |\n| 0.6221087710398319 | 0.6153961784334937  | 0.18428708381381365 | 0.624916705305911   | 20231023T144052Z     |\n| 0.4377277390071145 | 0.07538124164297655 | 0.04735527880151513 | 0.47809379567067456 | 20231023T144052Z     |\n| 0.8018721775350193 | 0.7887301429407455  | 0.5614330800633979  | 0.9820047415219545  | 20231023T144052Z     |\n| 0.9581393536837052 | 0.31683612216887125 | 0.329668445620915   | 0.12394270048696299 | 20231023T144052Z     |\n| 0.8759326347420947 | 0.5680986526260692  | 0.5029668331126184  | 0.1193808979262484  | 20231023T144052Z     |\n"`

(Note the enclosing quote characters and literal \n instead of line breaks).

We actually want this to be plain text, formatted so that it displays
nicely on a console.

Commit:
✨ Add `ONBUILD` to DaaP Python Base (#2256)

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>

tom-webber committed:
fix and update get_schema lambda (#2261)

* fix get schema
* use DataProductSchema to fetch schema

Commit:
:recycle: Move delete_all_element_version_data_files to versioning module

LavMatt commented on containers/daap-python-base/src/var/task/data_platform_paths.py:
```diff
@@ -198,25 +199,51 @@ def extract_timestamp_from_curated_path(string: str):
+    return element_prefixes
+
+
+def s3_recursive_delete(bucket_name: str, prefixes: list[str]) -> None:
```
You think this would be better in versioning as well?

mitchdawson1982 replied:
Agreed, now moved to the versioning module.

LavMatt commented on containers/daap-python-base/src/var/task/versioning.py:
```diff
@@ -8,10 +8,15 @@
+from data_platform_paths import (
+    DataProductConfig,
+    DataProductElement,
+    delete_all_element_version_data_files,
```
Don't want this import anymore as it's in versioning now

mitchdawson1982 replied:
Removed

LavMatt reviewed (commented):
couple very minor/pedantic comments

Commit:
🛂 Adds `ap-rshiny-notesbook` to ECR trust (#2265)

Signed-off-by: Jacob Woffenden <jacob.woffenden@digital.justice.gov.uk>

mitchdawson1982 committed:
Merge branch 'main' into dpl-1662-metadata-versioning

Commit:
:pushpin: Update python base image version

mitchdawson1982 committed:
Merge branch 'main' into dpl-1662-metadata-versioning
